gtsfm/utils/io.py

```diff
@@ -311,8 +381,16 @@ def write_images(gtsfm_data: GtsfmData, images: List[Image], save_dir: str) -> N
             qw, qx, qy, qz = iRw_quaternion
 
             f.write(f"{i} {qw} {qx} {qy} {qz} {tx} {ty} {tz} {i} {img_fname}\n")
-            # TODO: write out the points2d
-            f.write("TODO\n")
+
+            # write out points2d
+            for j in range(gtsfm_data.number_tracks()):
+                track = gtsfm_data.get_track(j)
+                for k in range(track.number_measurements()):
+                    # process each measurement
```

Comment:
nit: change to "write each measurement" ?

Reply:
Done